src/helpers.php

```diff
@@ -365,20 +365,25 @@ function old_empty_or_null($key, $empty_value = '')
 
 if (! function_exists('is_multidimensional_array')) {
     /**
-     * If any of the items inside a given array is an array, the array is considered multidimensional.
+     * Check if the array is multidimensional.
      *
-     * @param  array  $array
-     * @return bool
+     * If $strict is enabled, the array is considered multidimensional only if all elements of the array are arrays.
      */
-    function is_multidimensional_array(array $array)
+    function is_multidimensional_array(array $array, bool $strict = false): bool
```

Comment:
This has become messy. Any better way to do the same thing, that doesn't require so many conditionals and is clearer to read?

Reply:
If not... ok... I guess...